test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalSpecBuilder.java

```diff
@@ -204,10 +206,25 @@ public T systemProperty(String property, String value) {
         return cast(this);
     }
 
+    @Override
+    public T systemProperty(String key, Supplier<String> supplier) {
+        this.systemPropertyProviders.add(s -> Map.of(key, supplier.get()));
+        return cast(this);
+    }
+
+    public T systemProperty(SystemPropertyProvider systemPropertyProvider) {
```

Comment:
We should probably add this method to the `LocalSpecBuilder` interface if it's `public`.

Reply:
done